pandas/core/generic.py

```diff
@@ -319,6 +319,26 @@ def _as_manager(self, typ: str, copy: bool_t = True) -> Self:
         # fastpath of passing a manager doesn't check the option/manager class
         return self._constructor(new_mgr).__finalize__(self)
 
+    @classmethod
+    def _from_mgr(cls, mgr: Manager, axes: list[Index]) -> Self:
+        """
+        Construct a new object of this type from a Manager object and axes.
+
+        Parameters
+        ----------
+        mgr : Manager
+            Must have the same ndim as cls.
+        axes : list[Index]
+
+        Notes
+        -----
+        The axes must match mgr.axes, but are required for future-proofing
+        in the event that axes are refactored out of the Manager objects.
+        """
+        obj = cls.__new__(cls)
+        NDFrame.__init__(obj, mgr)
```

Comment:
We will need to discuss if this should make a shallow copy of the input mgr for CoW (cc @phofl). If we keep this consistent with the current behaviour of `DataFrame(mgr)`, we should add this:

```
if using_copy_on_write():
    mgr = mgr.copy(deep=False)
```

We had some discussion on the PR that introduced this to what extent this is actually needed: https://github.com/pandas-dev/pandas/pull/51239#discussion_r1102968580

Reply:
Since this should only be used internally I think we should be safe without the extra copy.

Reply:
> Since this should only be used internally I think we should be safe without the extra copy.

We could also internally have cases that require it. Essentially, whenever we would do something that boils down to `return self._constructor_from_mgr(self.mgr)` (i.e. a code path that simplifies to that) in some method, this is required. 

That should probably be considered a bug in our implementation, but so in https://github.com/pandas-dev/pandas/pull/51239#discussion_r1102973442 we went for the safe route for now. 

If we want to change that, maybe better to do that in a separate PR?

Reply:
> If we want to change that, maybe better to do that in a separate PR?

Yah I'd like to punt on this for the time being.

Reply:
I think we are in a state now that we should be able to rely on doing this on the manager level and not here. I think I am ok with not doing a shallow copy here